sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_shared/uploads.py

```diff
@@ -598,10 +598,11 @@ def read(self, size):
                     chunk = chunk.encode(self.encoding)
                 data += chunk
                 count += len(chunk)
+        # This means count < size and what's leftover will be returned in this call.
```

Comment:
Can you take a look at this change in Blob and see what it was about?

Reply:
https://github.com/Azure/azure-sdk-for-python/pull/17551

This is what it was from (bugfix for a infinite looping generator bug) so seems like it is a good thing to duplicate across other packages?

Reply:
Cool, yeah looks like a change that should come to Datalake.